symforce/slam/imu_preintegration/preintegrated_imu_measurements.h

```diff
@@ -0,0 +1,96 @@
+/* ----------------------------------------------------------------------------
+ * SymForce - Copyright 2022, Skydio, Inc.
+ * This source code is under the Apache 2.0 license found in the LICENSE file.
+ * ---------------------------------------------------------------------------- */
+
+#pragma once
+
+#include <Eigen/Dense>
+
+#include <sym/rot3.h>
+#include <sym/util/epsilon.h>
+
+namespace sym {
+
+/**
+ * Struct of Preintegrated IMU Measurements (not include the covariance of change in orientation,
+ * velocity, and position).
+ */
+template <typename Scalar>
+struct PIMNoCovariance {
+  using Vector3 = Eigen::Matrix<Scalar, 3, 1>;
+  using Matrix3 = Eigen::Matrix<Scalar, 3, 3>;
+
+  sym::Rot3<Scalar> DR;
+  Vector3 Dv;
+  Vector3 Dp;
+  Matrix3 DR_D_gyro_bias;
+  Matrix3 Dv_D_gyro_bias;
+  Matrix3 Dv_D_accel_bias;
+  Matrix3 Dp_D_gyro_bias;
+  Matrix3 Dp_D_accel_bias;
+  Vector3 accel_bias;
+  Vector3 gyro_bias;
+
+  double integrated_dt;
+
+  /**
+   * Initialize instance struct with accel_bias and gyro_bias and all other values
+   * zeroed out (scalars, vectors, and matrices) or set to the identity (DR).
+   */
+  PIMNoCovariance(const Vector3& accel_bias, const Vector3& gyro_bias);
+};
+
+/**
+ * Struct of preintegrated IMU measurements (using on manifold pre-integration).
+ */
+template <typename Scalar>
+struct PreintegratedImuMeasurements {
+  using Vector3 = typename PIMNoCovariance<Scalar>::Vector3;
+  using Matrix3 = typename PIMNoCovariance<Scalar>::Matrix3;
+  using Matrix9 = Eigen::Matrix<Scalar, 9, 9>;
+
+  PIMNoCovariance<Scalar> pim;
+  Matrix9 covariance;  // covariance of [DR.ToTangent(), Dv, Dp]
+
+  /**
+   * Initialize with given accel_bias and gyro_bias
+   */
+  PreintegratedImuMeasurements(const Vector3& accel_bias, const Vector3& gyro_bias);
+
+  /**
+   * Integrate a new measurement.
+   *
+   * Precondition:
+   *   *this was initialized by a constructor and has only been mutated by IntegrateMeasurement
+   *     (called in order on consecutive IMU measurements)
```

Comment:
I'm a fan of just making the necessary things private with accessors so this is enforced, I don't think I see a strong reason not to, and then we don't need this note

Reply:
Hmm. Fair enough. I considered that, and the only reason I didn't do it is because I didn't like the idea of writing all the accessors.

But I suppose that would make it easier to document what each of the fields is supposed to be right on the struct, so it might be worth it from that perspective.

Reply:
I think generally even if we don't make things private, we should document things as what they're intended to be, and don't need disclaimers like "this doesn't work if you set these arbitrarily instead of / after using the constructor".  I'd also be ok just leaving them public, documenting the fields in the struct, and getting rid of this note

Reply:
Should be able to remove these two lines now that things are private and it's impossible to violate these?  And then the rest isn't really "preconditions" anymore, it's just args

Reply:
Oh yeah. Good point. Removed and changed `Preconditions` to `Args`.